src/catalog/catalog_entry/dependency_set_catalog_entry.cpp

```diff
@@ -0,0 +1,335 @@
+#include "duckdb/catalog/catalog_entry/dependency_set_catalog_entry.hpp"
+#include "duckdb/catalog/catalog_entry/dependency_catalog_entry.hpp"
+#include "duckdb/catalog/dependency_list.hpp"
+#include "duckdb/catalog/catalog_entry/schema_catalog_entry.hpp"
+#include "duckdb/common/printer.hpp"
+#include "duckdb/catalog/dependency_manager.hpp"
+#include "duckdb/catalog/catalog.hpp"
+#include "duckdb/catalog/duck_catalog.hpp"
+
+namespace duckdb {
+
+DependencySetCatalogEntry::DependencySetCatalogEntry(DuckCatalog &catalog, DependencyManager &dependency_manager,
+                                                     CatalogType entry_type, const string &entry_schema,
+                                                     const string &entry_name)
+    : catalog(catalog), name(DependencyManager::MangleName(entry_type, entry_schema, entry_name)),
```

Comment:
Can we wrap all mangled names into a struct so the type system helps us out more, e.g.:

```cpp
struct MangledEntryName {
   string name;
};

struct MangledDependencyName {
   string name;
};
```